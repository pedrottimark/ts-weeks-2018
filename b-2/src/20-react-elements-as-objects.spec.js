const React = require('react')
const format = require('pretty-format')

const elementSymbol = Symbol.for('react.element')

// Add serializer which formats React element as object instead of markup.
expect.addSnapshotSerializer({
    print(val) {
        return format(val)
    },
    test(val) {
        return val && val.$$typeof === elementSymbol
    },
})

describe('React elements as objects', () => {
    it('represents an empty element with no props', () => {
        const type = 'br'
        const props = null
        const element = React.createElement(type, props)
        expect(element).toMatchSnapshot()
    })
    it('represents an empty element with two props', () => {
        const type = 'img'
        const props = {
            src: 'pretty.jpeg',
            title: 'pretty picture',
        }
        const element = React.createElement(type, props)
        expect(element).toMatchSnapshot()
    })
    it('represents an element with number as child', () => {
        const type = 'td'
        const props = null
        const child = 17
        const element = React.createElement(type, props, child)
        expect(element).toMatchSnapshot()
    })
    it('represents an element with text and elements as children', () => {
        const type = 'li'
        const props = null
        const element = React.createElement(
            type,
            props,
            'Just',
            React.createElement('strong', null, 'do'),
            ' it!'
        )
        expect(element).toMatchSnapshot()
    })
    it('escapes angle brackets in text', () => {
        const item = {
            completed: false,
            id: 1,
            text: '<script src="do-do.js"></script>',
        }
        const type = 'li'
        const props = {
            className: item.completed ? 'completed' : 'uncompleted',
        }
        const child = item.text
        const element = React.createElement(type, props, child)
        expect(element).toMatchSnapshot()
    })
})
