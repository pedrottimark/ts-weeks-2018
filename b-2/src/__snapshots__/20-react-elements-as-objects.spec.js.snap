// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React elements as objects escapes angle brackets in text 1`] = `
Object {
  "$$typeof": Symbol(react.element),
  "_owner": null,
  "_store": Object {},
  "key": null,
  "props": Object {
    "children": "<script src=\\"do-do.js\\"></script>",
    "className": "uncompleted",
  },
  "ref": null,
  "type": "li",
}
`;

exports[`React elements as objects represents an element with number as child 1`] = `
Object {
  "$$typeof": Symbol(react.element),
  "_owner": null,
  "_store": Object {},
  "key": null,
  "props": Object {
    "children": 17,
  },
  "ref": null,
  "type": "td",
}
`;

exports[`React elements as objects represents an element with text and elements as children 1`] = `
Object {
  "$$typeof": Symbol(react.element),
  "_owner": null,
  "_store": Object {},
  "key": null,
  "props": Object {
    "children": Array [
      "Just",
      Object {
        "$$typeof": Symbol(react.element),
        "_owner": null,
        "_store": Object {},
        "key": null,
        "props": Object {
          "children": "do",
        },
        "ref": null,
        "type": "strong",
      },
      " it!",
    ],
  },
  "ref": null,
  "type": "li",
}
`;

exports[`React elements as objects represents an empty element with no props 1`] = `
Object {
  "$$typeof": Symbol(react.element),
  "_owner": null,
  "_store": Object {},
  "key": null,
  "props": Object {},
  "ref": null,
  "type": "br",
}
`;

exports[`React elements as objects represents an empty element with two props 1`] = `
Object {
  "$$typeof": Symbol(react.element),
  "_owner": null,
  "_store": Object {},
  "key": null,
  "props": Object {
    "src": "pretty.jpeg",
    "title": "pretty picture",
  },
  "ref": null,
  "type": "img",
}
`;
